### PR TITLE
2304: /backport jdk23 should provide a more helpful message

### PR DESCRIPTION
**Pull request by zhaosongzs**
This patch is trying to improve the bot's replies of backport command.

1. If the user issued `/backport jdkX`, the bot would check if there is a `jdkX` branch in current repo, if so, the bot would tell the user that to target a backport to `jdkX` branch in current repo, he should issue `/backport :jdkX`.

2. When user specified branch name in the command but the branch doesn't exist, the bot will list 10 branch names in the target repo, the branch names are just sorted in reverse alphabetical order, so we can't guarantee they are the most relevant ones.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Issue
 * [SKARA-2304](https://bugs.openjdk.org/browse/SKARA-2304): /backport jdk23 should provide a more helpful message (**Bug** - P3)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/skara.git pull/1667/head:pull/1667` \
`$ git checkout pull/1667`

Update a local copy of the PR: \
`$ git checkout pull/1667` \
`$ git pull https://git.openjdk.org/skara.git pull/1667/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1667`

View PR using the GUI difftool: \
`$ git pr show -t 1667`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/skara/pull/1667.diff">https://git.openjdk.org/skara/pull/1667.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/skara/pull/1667#issuecomment-2204345515)

### CODE REVIEW

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/BackportCommand.java [hunk 1]
```diff
@@ -194,9 +198,19 @@ private HostedRepository getTargetRepo(PullRequestBot bot, String repoName, Prin
...
         var potentialTargetRepo = repoNameOptional.flatMap(forge::repository);
         if (potentialTargetRepo.isEmpty()) {
+            var branchNamesInCurrentRepo = bot.repo().branches().stream().map(HostedBranch::name).toList();
+            if (branchNamesInCurrentRepo.contains(repoName)) {
+                reply.println("There is a branch `" + repoName + "` in current repository `" + bot.repo().name() + "`.");
+                reply.println("To target a backport to this branch in current repository:");
+                reply.println("`/backport :" + repoName + "`");
+                reply.println();
+            }
             reply.println("The target repository `" + repoNameArg + "` is not a valid target for backports. ");
```
Comment (erikj79) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/BackportCommand.java:208]: I think this line should be first in the message, then followed by the branch message if there is a branch match.

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/BackportCommand.java [hunk 2]
```diff
@@ -194,9 +198,19 @@ private HostedRepository getTargetRepo(PullRequestBot bot, String repoName, Prin
 
         var potentialTargetRepo = repoNameOptional.flatMap(forge::repository);
         if (potentialTargetRepo.isEmpty()) {
+            var branchNamesInCurrentRepo = bot.repo().branches().stream().map(HostedBranch::name).toList();
+            if (branchNamesInCurrentRepo.contains(repoName)) {
+                reply.println("There is a branch `" + repoName + "` in current repository `" + bot.repo().name() + "`.");
+                reply.println("To target a backport to this branch in current repository:");
+                reply.println("`/backport :" + repoName + "`");
```
Comment (erikj79) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/BackportCommand.java:205]:
```suggestion
                reply.println("There is a branch `" + repoName + "` in the current repository `" + bot.repo().name() + "`.");
                reply.println("To target a backport to this branch in the current repository use:");
                reply.println("`/backport :" + repoName + "`");
```

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/BackportCommand.java [hunk 3]
```diff
@@ -208,6 +222,16 @@ private Branch getTargetBranch(String[] parts, int index, HostedRepository targe
         var targetBranchHash = targetRepo.branchHash(targetBranchName);
         if (targetBranchHash.isEmpty()) {
             reply.println("The target branch `" + targetBranchName + "` does not exist");
+            reply.print("List of target branches: ");
```
Comment (erikj79) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/BackportCommand.java:225]:
```suggestion
            reply.print("List of valid branches: ");
```

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/BackportCommand.java (hunk 1 above)
Comment (zhaosongzs) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/BackportCommand.java:208]: Sure, will fix it

**Review by erikj79 — APPROVED** [review]

**Comment by zhaosongzs** [comment]
Thanks for the review!
/integrate

**Pull request closed without merging**